pkg/controllers/node/health/controller.go

```diff
@@ -70,6 +72,20 @@ func (c *Controller) Reconcile(ctx context.Context, node *corev1.Node) (reconcil
 		return reconcile.Result{}, nodeutils.IgnoreNodeClaimNotFoundError(err)
 	}
 
+	nodePoolName, found := nodeClaim.Labels[v1.NodePoolLabelKey]
+	if !found {
+		return reconcile.Result{}, fmt.Errorf("nodepool cannot be found from node")
```

Comment:
This is going to requeue -- this shouldn't requeue IMO since this isn't recoverable -- we should perhaps think about how this is going to work for standalone NodeClaims in the future since this is going to remove support for those -- maybe something with just not considering percentage when they are standalone, or doing the whole cluster?